src/utils/daos_hdlr.c

```diff
@@ -1949,27 +1949,41 @@ dm_connect(struct cmd_args_s *ap,
 			rc = -DER_NONEXIST;
 		}
 		if (rc == -DER_NONEXIST) {
-			uuid_t cuuid;
+			uuid_t	cuuid;
+			bool	dst_cont_is_uuid = true;
+
+			if (dst_cont_passed) {
+				rc = uuid_parse(ca->dst_cont, cuuid);
+				dst_cont_is_uuid = (rc == 0);
+			}
 
 			if (ca->cont_layout == DAOS_PROP_CO_LAYOUT_POSIX) {
 				attr.da_props = props;
-				rc = dfs_cont_create(ca->dst_poh, &cuuid, &attr, NULL, NULL);
-				if (rc != 0) {
+				if (dst_cont_is_uuid)
+					rc = dfs_cont_create(ca->dst_poh, &cuuid, &attr,
+							     NULL, NULL);
+				else
+					rc = dfs_cont_create_with_label(ca->dst_poh, ca->dst_cont,
+									&attr, &cuuid, NULL, NULL);
+				if (rc) {
```

Comment:
This pattern is copied from mpifileutils, though I'm not sure if it's really how we should be doing this.